src/extension/tab/hook.ts

```diff
@@ -94,20 +94,21 @@ function initializeHook() {
   function sendHookDataToDevTools(eventName: "update" | "connectToDevtools") {
     tab.send({
       type: eventName,
-      payload: {
-        queries: hook.getQueries(),
-        mutations: hook.getMutations(),
-
-        // We need to JSON stringify the cache here in case the cache contains
-        // references to `URL` instances which are not cloneable via
-        // `structuredClone` (which `window.postMessage` uses to send messages).
-        // `JSON.stringify` does however serialize `URL`s into strings properly,
-        // so this should ensure that the cache data will be sent without
-        // errors.
-        //
-        // https://github.com/apollographql/apollo-client-devtools/issues/1258
-        cache: JSON.stringify(hook.getCache()),
-      },
+      // We need to JSON stringify the data here in case the cache contains
+      // references to irregular data such as `URL` instances which are not
+      // cloneable via `structuredClone` (which `window.postMessage` uses to
+      // send messages). `JSON.stringify` does however serialize `URL`s into
+      // strings properly, so this should ensure that the cache data will be
+      // sent without errors.
+      //
+      // https://github.com/apollographql/apollo-client-devtools/issues/1258
+      payload: JSON.parse(
```

Comment:
I'll note here that I think a potential future improvement here is to catch errors in case the cache can't be fully stringified. It would be a much nicer experience to show an error in the devtools panel, but we'll need to figure out a way to transport that error over. That is a bit out of scope for this PR.